pkgs/by-name/kp/kplex/package.nix

```
@@ -0,0 +1,27 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  lib,
+}:
+stdenv.mkDerivation rec {
+  pname = "kplex";
+  version = "1.4";
+  src = fetchFromGitHub {
+    owner = "stripydog";
+    repo =  "kplex";
+    rev = "v${version}";
```

Consider using the `finalAttrs` pattern. It makes it a bit easier to override the package. (You'll have to add the last parenthesis as well)

```suggestion
stdenv.mkDerivation (finalAttrs: {
  pname = "kplex";
  version = "1.4";
  src = fetchFromGitHub {
    owner = "stripydog";
    repo =  "kplex";
    rev = "v${finalAttrs.version}";
```

---

Does mkDerivation do a fixed-point evaluation of the around finalAttrs? I can't find any documentation for this

---

Yeah, I think so. It's *relatively* new, but I think the main intention is to not have to override multiple attributes when doing `package.override { version = ... }`. This PR contains a lot of information and relevant links if you're interested: #234651